File: java/datevalidation.java
package com.selenium;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.concurrent.TimeUnit;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.ie.InternetExplorerDriver;
import org.testng.annotations.Test;


public class datevalidation {
	WebDriver driver;
	
	@Test
	public void datevalidation() throws InterruptedException{
		String browser = "Mozilla";// xls, xml
		// script
		if(browser.equals("Mozilla")){
			driver = new FirefoxDriver();
		}else if(browser.equals("Chrome")){
			driver = new ChromeDriver();
		}else if(browser.equals("IE")){
			driver = new InternetExplorerDriver();
		}else if(browser.equals("Edge")){
			driver = new EdgeDriver();
		}
		
		driver.manage().timeouts().implicitlyWait(20,TimeUnit.SECONDS);
		driver.manage().timeouts().pageLoadTimeout(60,TimeUnit.SECONDS);
		driver.manage().window().maximize();
		driver.get("https://www.seleniumeasy.com/test/bootstrap-date-picker-demo.html");
		driver.findElement(By.xpath("//html/body/div[2]/div/div[2]/div[1]/div/div[2]/div/div/span/i")).click();
		selectDate("21/08/2020");
		// wait 

	}

	public void selectDate(String d){
		// day, month , year
		Date current = new Date();
		SimpleDateFormat sd = new SimpleDateFormat("dd/MM/yyyy");
		try {
			Date selected = sd.parse(d);
			String day = new SimpleDateFormat("dd").format(selected);
			String month = new SimpleDateFormat("MMMM").format(selected);
			String year = new SimpleDateFormat("yyyy").format(selected);
			System.out.println(day+" --- "+month+" --- "+ year);
			String desiredMonthYear=month+" "+year;
			
			while(true){
				String displayedMonthYear=driver.findElement(By.cssSelector("th.datepicker-switch")).getText();
				if(desiredMonthYear.equals(displayedMonthYear)){
					// select the day
					driver.findElement(By.xpath("//td[text()='"+day+"']")).click();
					break;
				}else{
					
					if(selected.compareTo(current) > 0)
						driver.findElement(By.xpath("//html/body/div[3]/div[1]/table/thead/tr[2]/th[3")).click();
					else if(selected.compareTo(current) < 0)
						driver.findElement(By.xpath("//html/body/div[3]/div[1]/table/thead/tr[2]/th[1]")).click();

				}
			}

		} catch (ParseException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}
	
}

File: datevalidation.py
'''
Created on 6 Oct 2020

@author: FMORASSU
'''

from selenium import webdriver
from selenium.webdriver.support.select import Select
from datetime import datetime

#op = webdriver.ChromeOptions()
#op.add_argument("user-data-dir=C:\\Users\\FMORASSU\\AppData\\Local\\Google\\Chrome\\User Data\\Default\\Profile 1")
#op.add_argument("--disable-bookmarks")
#op.add_argument("--disable-notifications")
#op.add_argument("--start-maximized")
#op.add_argument("--ignore-certificate-errors")

#driver = webdriver.Chrome(options=op)

def selectDate(d):
    currentDate = datetime.now()
    print(currentDate.strftime('%d/%m/%Y'))
    dt = datetime.strptime(d, '%d/%m/%Y')
    Year=dt.year
    day=dt.day
    month=dt.strftime("%B")
    print(Year)
    print(day)
    print(month)
    
    desirMY=str(month)+" " +str(Year)
    print(desirMY)
    
    while True:
        #dispMY = driver.find_element_by_xpath("//html/body/div[3]/div[1]/table/thead/tr[2]/th[2]").text
        dispMY = driver.find_element_by_css_selector("th.datepicker-switch").text
        print(dispMY)
        if(desirMY==dispMY):
            driver.find_element_by_xpath("//td[text()= '"+str(day)+"']").click()
            break
        else:
            if(dt>currentDate):
                driver.find_element_by_xpath("//html/body/div[3]/div[1]/table/thead/tr[2]/th[3").click()
            elif(dt<currentDate):
                driver.find_element_by_xpath("//html/body/div[3]/div[1]/table/thead/tr[2]/th[1]").click()
                 
browseroption = "Chrome"
webdriver.driver = None


if browseroption=='Chrome':
    driver = webdriver.Chrome()
elif browseroption=='Firefox':
    driver = webdriver.Firefox()
elif browseroption=='Edge':
    driver = webdriver.Edge()
elif browseroption=='IE':
    driver = webdriver.Ie()
else:
    print("No browser specified")


driver.maximize_window()    
driver.get("https://www.seleniumeasy.com/test/bootstrap-date-picker-demo.html")
driver.implicitly_wait(5)


driver.find_element_by_xpath("//html/body/div[2]/div/div[2]/div[1]/div/div[2]/div/div/span/i").click()
selectDate("21/09/2020")
